packages/front-end/components/Settings/SecretApiKeys.tsx

```diff
@@ -14,103 +15,202 @@ const SecretApiKeys: FC<{ keys: ApiKeyInterface[]; mutate: () => void }> = ({
-                      getValue={async () => {
-                        const res = await apiCall<{ key: SecretApiKey }>(
-                          `/keys/reveal`,
-                          {
-                            method: "POST",
-                            body: JSON.stringify({
-                              id: key.id,
-                            }),
-                          }
-                        );
-                        if (!res.key.key) {
-                          throw new Error("Could not load the secret key");
-                        }
-                        return res.key.key;
-                      }}
-                    />
-                  ) : (
-                    <em>hidden</em>
-                  )}
-                </td>
-                {canManageKeys && (
-                  <td>
-                    <MoreMenu>
-                      <DeleteButton
-                        onClick={async () => {
-                          await apiCall(`/keys`, {
-                            method: "DELETE",
-                            body: JSON.stringify({
-                              id: key.id,
-                            }),
-                          });
-                          mutate();
-                        }}
-                        className="dropdown-item"
-                        displayName="Secret Api Key"
-                        text="Delete key"
-                      />
-                    </MoreMenu>
-                  </td>
-                )}
-              </tr>
-            ))}
-          </tbody>
-        </table>
-      )}
-      {canManageKeys && (
-        <button
-          className="btn btn-primary"
-          onClick={(e) => {
-            e.preventDefault();
-            setOpen(true);
-          }}
-        >
-          <FaKey /> Create New Secret Key
-        </button>
-      )}
+
+      {/* region Secret API keys */}
+      <div className="mb-5">
+        <h1>Secret API Keys</h1>
+        <p>
+          Secret keys have full read and write access to your organization.
+          Because of this, they must be kept secure and{" "}
+          <strong>must not be exposed to users</strong>.
+        </p>
+        {secretKeys.length > 0 && (
+          <ApiKeysTable
+            onDelete={onDelete}
+            keys={secretKeys}
+            canManageKeys={canManageKeys}
+            onReveal={onReveal}
+          />
+        )}
+        {canManageKeys && (
+          <button
+            className="btn btn-primary"
+            onClick={(e) => {
+              e.preventDefault();
+              // Secret API keys have no type
+              setModalApiKeyType(undefined);
+              setOpen(true);
+            }}
+          >
+            <FaKey /> Create New Secret Key
+          </button>
+        )}
+      </div>
+      {/* endregion Secret API keys */}
+
+      {/* region User API keys */}
+      <div className="mb-5">
+        <h1>User API Keys</h1>
```

Comment:
The UI doesn't quite feel right here.  I'd expect every user to be able to manage their own Personal Access Tokens, even if they are not an admin and missing the `manageApiKeys` permission.  As such, I'd expect it to be accessible from the user dropdown in the top nav, not under Settings.

The **Settings -> API Keys** page will be just for admins and contain only the secret keys.  I think this will be much less confusing for users.

Reply:
As suggested, I've added the personal access tokens to a new page (under settings but not in the settings menu) under the user menu.